compile.ts

```diff
@@ -0,0 +1,175 @@
+    } else {
+      return `${messageText}\n${brief}\n`;
+    }
+  }
+}
+
+async function emit(file: string): Promise<{
+  diagnostics: Deno.Diagnostic[];
+  files: Record<string, string>;
+}> {
+  const { diagnostics, files } = await Deno.emit(file, {
+    compilerOptions: {
+      declaration: true,
+      sourceMap: true,
+      target: "es2020",
+      module: "esnext",
+      lib: [
+        "dom",
+        "DOM.Iterable",
+        "esnext",
+      ],
+      removeComments: true,
+      downlevelIteration: true,
+      useDefineForClassFields: true,
+      strict: true,
+      noImplicitReturns: true,
+      noFallthroughCasesInSwitch: true,
+      importsNotUsedAsValues: "error",
+      paths: {
+        "/dist/*": ["src/*"],
+      },
+      baseUrl: "./",
+    },
+  });
+  return {
+    diagnostics,
+    files,
+  };
+}
+
+/**
+ * Write a file to the dist directory. Replaces the following in the `filename`:
+ *   - `src` with `dist`
+ *   - `.ts.d.ts` with `.d.ts`
+ *   - `.ts.js` with `.js`
+ *
+ * @param filename The file url of an emitted file. Must be the result of `emit`
+ * @param fileContent - The file content associated with `files[filename]`
+ */
+function write(filename: string, fileContent: string): void {
+  const outPath = filename.replace("src", "dist").replace(".ts.js", ".js")
+    .replace(".ts.d.ts", ".d.ts");
+  if (outPath.endsWith(".d.ts") || outPath.endsWith(".js")) { // ensure directory(s) file is in, exists
+    const pathSplit = outPath.split("/");
+    pathSplit.pop();
+    const parentDirOfFile = pathSplit.join("/");
+    const validPath = Deno.build.os === "windows"
+      ? fromFileUrlWin(parentDirOfFile)
+      : parentDirOfFile; // because `ensureDirSync` will throw an error if the path is a file url on windows
+    ensureDirSync(validPath);
+  }
+
+  // Because imports inside the files are still using a .ts extension, we're going to make them use .js:
+  fileContent = fileContent.replace(
+    /import { ([a-zA-Z].*) } from "(.*)";/gm,
+    (_str, importValues, fileImportedFrom) => {
+      const jsImport = fileImportedFrom.replace(".ts", ".js");
+      return `import { ${importValues} } from \"${jsImport}\";`;
+    },
+  );
+  fileContent = fileContent.replace(
+    /export { ([a-zA-Z].*) } from "(.*)";/gm,
+    (_str, importValues, fileImportedFrom) => {
+      const jsImport = fileImportedFrom.replace(".ts", ".js");
+      return `export { ${importValues} } from \"${jsImport}\";`;
+    },
+  );
+  fileContent = fileContent.replace(
+    /import \"(.*)\";/gm,
+    (_str, importValue) => {
+      const jsRef = importValue.replace(".ts", ".js");
+      return `import \"${jsRef}\";`;
+    },
+  );
+  fileContent = fileContent.replace(
+    /import\(\"(.*)\"\)/gm,
+    (_str, importValue) => {
+      const jsRef = importValue.replace(".ts", ".js");
+      return `import(\"${jsRef}\")`;
+    },
+  );
+  /// <amd-module name="file:///D:/Development/ebebbington/destiny/src/mod.ts" />
```

Comment:
hopefully this whole section (add example imports maybe in mod.ts) would be resolved by https://discord.com/channels/684898665143206084/689420767620104201/849740669059727390